src/utils.ts

```diff
@@ -9,3 +11,15 @@ export function debounce(fn, wait = 0) {
         }, wait);
     };
 }
+
+export const mapSettingsAreEqual = (mapSettings1: MapSettings, mapSettings2: MapSettings) => {
+    if (mapSettings1 === null && !!mapSettings2) return false;
+    if (mapSettings2 === null && !!mapSettings1) return false;
+    return (
+        mapSettings1.pathogen === mapSettings2.pathogen &&
+        mapSettings1.version === mapSettings2.version &&
+        mapSettings1.indicator === mapSettings2.indicator &&
+        mapSettings1.country === mapSettings2.country &&
+        mapSettings1.adminLevel === mapSettings2.adminLevel
+    );
+};
```

Comment:
That might not be ideal, it's just an example

Reply:
yhhh i like this! there is something to be said about using one of them as the source for the keys but assuming they are both typed correctly as `MapSettings` this is nice

Reply:
pretty sure that is not possible, since types are stripped away when typescript is transpiled into base javascript, we cant have any javascript implementation dependent on a type

Reply:
Good to know!